relay-server/src/services/processor/profile.rs

```diff
@@ -94,6 +94,25 @@ pub fn transfer_id(event: &mut Annotated<Event>, profile_id: Option<ProfileId>)
     }
 }
 
+pub fn scrub_profiler_id(event: &mut Annotated<Event>) {
+    let Some(event) = event.value_mut() else {
+        return;
+    };
+    let transaction_duration = event
+        .get_value("event.duration")
+        .and_then(|duration| duration.as_f64());
+
+    if let Some(contexts) = event.contexts.value_mut() {
+        if let Some(profile_context) = contexts.get_mut::<ProfileContext>() {
+            if profile_context.profiler_id.value().is_some()
+                && transaction_duration.is_some_and(|duration| duration < 20_f64)
+            {
+                profile_context.profiler_id = Annotated::empty();
+            }
+        }
+    }
```

Comment:
Let's annotate this properly, this should work:

```suggestion
    if !transaction_duration.is_some_and(|duration| duration < 20f64) {
        return;
    }

    if let Some(contexts) = event.contexts.value_mut().as_mut() {
        if let Some(profiler_id) = contexts
            .get_mut::<ProfileContext>()
            .map(|ctx| &mut ctx.profiler_id)
        {
            let id = std::mem::take(profiler_id.value_mut());
            let remark = Remark::new(RemarkType::Removed, "transaction_duration");
            profiler_id.meta_mut().add_remark(remark);
            profiler_id.meta_mut().set_original_value(id);
        }
    }
```